dnf5-plugins/config-manager_plugin/addrepo.cpp

```diff
@@ -0,0 +1,462 @@
+                    throw ConfigManagerError(
+                        M_("Sets the \"{}\" option again with a different value: \"{}\" != \"{}\""),
+                        key,
+                        it->second,
+                        key_value);
+                }
+            }
+            return true;
+        });
+    cmd.register_named_arg(set_opt);
+
+    // Set conflicting arguments
+    repo_id_opt->add_conflict_argument(*from_repofile_opt);
+    set_opt->add_conflict_argument(*from_repofile_opt);
+}
+
+
+void ConfigManagerAddRepoCommand::configure() {
+    auto & ctx = get_context();
+    auto & base = ctx.base;
+    auto logger = base.get_logger();
+
+    const auto & repo_dirs = base.get_config().get_reposdir_option().get_value();
+    if (repo_dirs.empty()) {
+        throw ConfigManagerError(M_("Missing path to repository configuration directory"));
+    }
+
+    std::filesystem::path repo_dir = repo_dirs.front();
+
+    std::string url;
+    if (!repofile_url.empty()) {
+        url = repofile_url;
+    } else if (auto it = repo_opts.find("baseurl"); it != repo_opts.end() && !it->second.empty()) {
+        const auto urls = libdnf5::OptionStringList(std::vector<std::string>{}).from_string(it->second);
+        if (urls.empty()) {
+            throw ConfigManagerError(M_("Bad baseurl: {}={}"), it->first, it->second);
+        }
+        url = urls[0];
+    }
+    if (url.empty()) {
+        if (auto it = repo_opts.find("mirrorlist"); it != repo_opts.end() && !it->second.empty()) {
+            url = it->second;
+        } else if (auto it = repo_opts.find("metalink"); it != repo_opts.end() && !it->second.empty()) {
+            url = it->second;
+        } else {
+            throw ArgumentParserMissingDependentArgumentError(
+                M_("One of --from-repofile=<URL>, --set=baseurl=<URL>, --set=mirrorlist=<URL>, --set=metalink=<URL> "
+                   "must be set to a non-empty URL"));
+        }
+    }
+
+    if (!repofile_url.empty()) {
+        add_repo_from_repofile(url, repo_dir);
+    } else {
+        create_repo(url, repo_dir);
+    }
+}
+
+
+void ConfigManagerAddRepoCommand::add_repo_from_repofile(
+    const std::string & url, const std::filesystem::path & repo_dir) {
+    auto & ctx = get_context();
+    auto & base = ctx.base;
+    auto logger = base.get_logger();
+
+    logger->info("config-manager: Adding repofile from: {}", url);
+
+    std::string full_url;
+    if (get_url_part(url, CURLUPART_SCHEME) == "") {
+        full_url += "file://" + std::filesystem::absolute(url).string();
+    } else {
+        full_url = url;
+    }
+
+    if (save_filename.empty()) {
+        save_filename = std::filesystem::path(get_url_part(full_url, CURLUPART_PATH)).filename();
+    }
+    if (!save_filename.ends_with(".repo")) {
+        save_filename += ".repo";
+    }
+    auto dest_path = repo_dir / save_filename;
+
+    test_filepath(dest_path);
+
+    auto tmpfilepath = dest_path.string() + ".XXXXXX";
```

Comment:
May I ask you to add a comment what the  the code does between line 301 and 307. It looks like that it only check whether it is possible to create a temp file.

Reply:
No. The code will produce an open temporary file. It then closes it but does not remove it. In the following code, this temporary file is used to store the copied/downloaded configuration.